### PR TITLE
docs: generalize mac-specific sections of first PR quickstart

### PR DESCRIPTION
**Pull request by kdmccormick**

### CODE REVIEW

File: source/developers/quickstarts/first_openedx_pr.rst [hunk 1]
```diff
@@ -34,30 +34,35 @@ Before you start
    For the smoothest experience, we recommend that your computer has at least
    **16 GB of RAM, 2 CPUs, and at least 50 GB of Free disk space**.
 
-This tutorial is written for users of macOS with either an Intel or an ARM
-(Apple Silicon, M1, M2, etc.) processor or Linux (x86 or x86_64). Additionally,
+This tutorial is written for users of macOS or Linux.
+It has been tested with both common 64-bit processor families:
+x86-64 (a.k.a. Intel-64) and ARM-64 (used in newer Apple laptops, including M1, M2, etc.).
```
Comment (kdmccormick) [inline source/developers/quickstarts/first_openedx_pr.rst:39]: I reworded this a bit to capture that:
* [tutor doesn't guarantee 32-bit support](https://docs.tutor.overhang.io/install.html)
* there's no reason the ARM-64 instructions shouldn't also work on Linux (🤞🏻) although I avoided explicitly saying it would work on such a system

File: source/developers/quickstarts/first_openedx_pr.rst [hunk 2]
```diff
@@ -34,30 +34,35 @@ Before you start
...
 you will need to have the following installed or configured, and know at least
 the basics of using them, before proceeding:
 
 * Git (see `GitHub's set up Git guide
   <https://help.github.com/en/github/getting-started-with-github/set-up-git>`_)
 
 * A `GitHub account <https://github.com/signup>`_
-  * Additionally, we strongly recommend setting up 2-factor authentication
+
+  * Additionally, we strongly recommend setting up 2-factor authentication.
```
Comment (kdmccormick) [inline source/developers/quickstarts/first_openedx_pr.rst:49]: unrelated formatting fix

File: source/developers/quickstarts/first_openedx_pr.rst [hunk 3]
```diff
@@ -34,30 +34,35 @@ Before you start
...
+  `MacVim <https://github.com/macvim-dev/macvim>`_,
+  and others)
 
 .. _homebrew: https://brew.sh
 
 Install Docker
 **************
 
-Install `Docker for macOS <https://docs.docker.com/docker-for-mac/>`_ and launch
+Install `Docker Desktop <https://docs.docker.com/desktop/>`_ and launch
```
Comment (kdmccormick) [inline source/developers/quickstarts/first_openedx_pr.rst:65]: docker has a nice landing page here that lets you pick macOS, Linux, or Windows

**Review by kdmccormick — COMMENTED** [review]
Notes for reviewers

**Comment by kdmccormick** [comment]
@feanil or @Carlos-Muniz , here are a couple more tweaks to the quickstart.

**Review by feanil — APPROVED** [review]
